qa/ingest-with-mustache/src/test/java/org/elasticsearch/ingest/IngestDocumentMustacheIT.java

```diff
@@ -44,6 +44,30 @@ public void testAccessMetaDataViaTemplate() {
         assertThat(ingestDocument.getFieldValue("field1", String.class), equalTo("2 bar"));
     }
 
+    // TODO(talevy): expected?
+    public void testAccessSubFieldsViaTemplate() {
+        Map<String, Object> document = new HashMap<>();
+        Map<String, Object> fooDoc = new HashMap<>();
+        fooDoc.put("bar", "value");
+        document.put("foo", fooDoc);
+        IngestDocument ingestDocument = new IngestDocument("index", "type", "id", null, null, null, null, document);
+        ingestDocument.setFieldValue(templateService.compile("field1"), ValueSource.wrap("1 {{foo.bar}}", templateService));
+        assertThat(ingestDocument.getFieldValue("field1", String.class), equalTo("1 value"));
+    }
+
+    // TODO(talevy): expected?
+    public void testAccessFieldWithDotsViaTemplate() {
+        Map<String, Object> document = new HashMap<>();
+        Map<String, Object> fooBarDoc = new HashMap<>();
+        fooBarDoc.put("val", "value");
+        document.put("foo.bar", fooBarDoc);
+        IngestDocument ingestDocument = new IngestDocument("index", "type", "id", null, null, null, null, document);
+        ingestDocument.setFieldValue(templateService.compile("field1"), ValueSource.wrap("1 {{foo.bar}}", templateService));
+        assertThat(ingestDocument.getFieldValue("field1", String.class), equalTo("1 {val=value}"));
```

Comment:
at least mustache returns something, while IngestDocument.getFieldValue("foo.bar") throws an exception I think, cause it splits the path at the dots. and looks for bar within foo. mustache instead seems to return a map entry (key=value). Not sure what we would want it to return instead?